src/mlpack/methods/softmax_regression/softmax_regression_impl.hpp

```diff
@@ -17,118 +17,170 @@
 
 namespace mlpack {
 
-template<typename OptimizerType>
-SoftmaxRegression::SoftmaxRegression(
-    const arma::mat& data,
-    const arma::Row<size_t>& labels,
+template<typename MatType>
+inline SoftmaxRegression<MatType>::SoftmaxRegression(
+    const size_t inputSize,
     const size_t numClasses,
-    const double lambda,
-    const bool fitIntercept,
-    OptimizerType optimizer) :
+    const bool fitIntercept) :
     numClasses(numClasses),
-    lambda(lambda),
+    lambda(0.0001),
```

Comment:
`lambda` is always `double` regardless of `MatType` (just for convenience and clarity), so I don't think it's needed here... let me know if you think otherwise. :+1: